test/influx.test.js

```diff
@@ -1,58 +1,131 @@
+import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest'
+import { createMetricsFromRequest, writeMetrics, reportRequestMetric } from '../lib/influx.js'
 
-describe('reportMetric', () => {
-  it('should report metrics correctly', async () => {
-    const request = {
-      url: 'https://example.com/path?api-key=test-key',
-      method: 'GET',
-      headers: new Map([
-        ['api-key', 'test-key'],
-        ['Authorization', 'Bearer test-token']
-      ])
-    };
+describe('reportRequestMetric', () => {
+  const date = new Date()
+
+  beforeEach(() => {
+    vi.useFakeTimers()
+    vi.setSystemTime(date)
+  })
 
-    const env = {
-      INFLUX_METRIC_NAME: 'test_metric',
-      INFLUX_URL: 'https://influx.example.com',
-      INFLUX_DATABASE: 'test_db',
-      INFLUX_TOKEN: 'test_token'
-    };
+  afterEach(() => {
+    vi.useRealTimers()
+  })
 
-    global.fetch = vi.fn().mockResolvedValue({ status: 204 });
+  it('reports request metrics to InfluxDB over HTTP', async () => {
+    const env = withTestEnvitronment()
+    const request = {
+      url: 'https://example.com/path',
+      method: 'GET',
+      headers: new Map([['api-key', 'test-key']]),
+    }
+    global.fetch = vi.fn().mockResolvedValue(new Response(null, { status: 204 }));
 
-    await reportMetric(request, env);
+    await reportRequestMetric(request, env);
 
     expect(global.fetch).toHaveBeenCalledWith(
       'https://influx.example.com/api/v2/write?&bucket=test_db&precision=ms',
       expect.objectContaining({
         method: 'POST',
         headers: expect.objectContaining({
-          'Authorization': 'Token test_token',
-          'Content-Type': 'application/octet-stream'
-        })
-      })
-    );
+          Authorization: 'Token test_token',
+          'Content-Type': 'application/octet-stream',
+        }),
+        body: `test_metric api_key="test-key" ${date.getTime()}`,
+      }),
+    )
   });
 });
 
 describe('createMetricsFromRequest', () => {
-  it('should format request data correctly', () => {
-    const request = {
-      url: 'https://example.com/path?api-key=test-key',
-      method: 'GET',
-      headers: new Map([
-        ['api-key', 'test-key'],
-        ['Authorization', 'Bearer test-token']
-      ])
-    };
-
-    const env = {
-      INFLUX_METRIC_NAME: 'test_metric'
-    };
-
-    const result = createMetricsFromRequest(request, env);
-    expect(result).toContain('test_metric');
-    expect(result).toContain('api_key="test-key"');
-  });
-});
\ No newline at end of file
+  const date = new Date()
+  const env = withTestEnvitronment()
+
+  beforeEach(() => {
+    vi.useFakeTimers()
+    vi.setSystemTime(date)
+  })
+
+  afterEach(() => {
+    vi.useRealTimers()
+  })
+
+  const testCases = [
+    {
+      description: 'when request has an API key in the Authorization header',
+      request: {
+        url: 'https://example.com/path',
+        method: 'GET',
```

Comment:
Let's omit these two fields from the test case definitions, since they are the same in all cases, and fill them inside the test.

```
testCases.forEach(({ description, request, expectedApiKey }) => {
    it(description, () => {
      const fullRequest = {
         url: 'https://example.com/path',
         method: 'GET',
         headers: new Map(),
         ...request
      }
      const result = createMetricsFromRequest(fullRequest, env)
      expect(result).toContain(expectedApiKey)
    })
  })
```